src/Permissions.ts

```diff
@@ -56,56 +56,73 @@ export class Permissions {
       invitationIdsCreatedByUserId: string[];
     }
   ) {
-    let canI = false;
 
+    const role = this.getRole(forSpace, forUserId);
+    const isAdminOrOwner = [Roles.Owner, Roles.Admin].includes(role);
+
+    const canPatchOrRevokeInvitation = () => {
+      if (isAdminOrOwner) {
+        return true;
+      }
+
+      if (
+        // If there is an invitationId to be revoked
+        forRevokeInvitation?.invitationId
+        // If this user has created more than one invitation
+        && forRevokeInvitation?.invitationIdsCreatedByUserId?.length > 0
+        // If invitation to revoke was created by userId
+        && forRevokeInvitation?.invitationIdsCreatedByUserId.find(
+          invitationIdCreatedByUserId => invitationIdCreatedByUserId === forRevokeInvitation?.invitationId
+        )
+      ) {
+        return true;
+      }
+
+      return false;
+    }
     switch (action) {
       case Actions.ChangeSpacePlan:
-      case Actions.DeleteSpace:
-        canI = forSpace.kind !== "Personal" && this.getRole(forSpace, forUserId) === Roles.Owner;
-        break;
-      case Actions.PatchInvitation:
-      case Actions.RevokeInvitation: {
-        if ([Roles.Owner, Roles.Admin].includes(this.getRole(forSpace, forUserId))) {
-          canI = true;
-        }
+        return role === Roles.Owner
```

Comment:
`;` missing

Reply:
yeah, vs code doesn't fix lint on this repo...